Week 04/Lecture 08/Assignment 03/employee_manager/src/main/java/com/example/employee_manager/config/DataSourceConfig.java

```diff
@@ -0,0 +1,21 @@
+package com.example.employee_manager.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+
+import javax.sql.DataSource;
+
+@Configuration
+public class DataSourceConfig {
+
+    @Bean
+    public DataSource getDataSource() {
+        DataSourceBuilder<?> dataSourceBuilder = DataSourceBuilder.create();
+        dataSourceBuilder.driverClassName("com.mysql.cj.jdbc.Driver");
```

Comment:
Database info should put in prop file and get by @Value. shoud not hardcode like this

Reply:
Thank you for the feedback, I have resolved that data source configuration, please check my latest commit. I've added like 
```java
@Configuration
public class DataSourceConfig {

    @Value("${spring.datasource.primary.url}")
    private String primaryUrl;
 
    // ... other code
```